selfdrive/manager/process.py

```diff
@@ -0,0 +1,225 @@
+import importlib
+import os
+import signal
+import time
+import subprocess
+from abc import ABC, abstractmethod
+from multiprocessing import Process
+
+from setproctitle import setproctitle  # pylint: disable=no-name-in-module
+
+import cereal.messaging as messaging
+import selfdrive.crash as crash
+from common.basedir import BASEDIR
+from common.params import Params
+from selfdrive.swaglog import cloudlog
+from selfdrive.hardware import HARDWARE
+from cereal import log
+
+
+def launcher(proc):
+  try:
+    # import the process
+    mod = importlib.import_module(proc)
+
+    # rename the process
+    setproctitle(proc)
+
+    # create new context since we forked
+    messaging.context = messaging.Context()
+
+    # exec the process
+    mod.main()
+  except KeyboardInterrupt:
+    cloudlog.warning("child %s got SIGINT" % proc)
+  except Exception:
+    # can't install the crash handler becuase sys.excepthook doesn't play nice
+    # with threads, so catch it here.
+    crash.capture_exception()
+    raise
+
+
+def nativelauncher(pargs, cwd):
+  # exec the process
+  os.chdir(cwd)
+  os.execvp(pargs[0], pargs)
+
+
+def join_process(process, timeout):
+  # Process().join(timeout) will hang due to a python 3 bug: https://bugs.python.org/issue28382
+  # We have to poll the exitcode instead
+  t = time.time()
+  while time.time() - t < timeout and process.exitcode is None:
```

Comment:
```suggestion
  t = time.monotonic()
  while time.monotonic() - t < timeout and process.exitcode is None:
```

Reply:
Maybe with python 3.8 this whole function can go. Will look into that after merging.